dvc/repo/reproduce.py

```diff
@@ -117,37 +87,15 @@ def _reproduce(
     stages = nx.get_node_attributes(G, "stage")
     node = relpath(stage.path, self.root_dir)
 
-    if single_item:
-        ret = _reproduce_stage(
-            stages, node, force, dry, interactive, no_commit
-        )
+    if kwargs.get("single_item", False):
```

Comment:
You can just `def _reproduce(self, target, single_item=False, **kwargs)`, since it is not used anywhere down the line and there is no need to pass it through. Take a look at other places too.